gtsam/nonlinear/NonlinearFactorGraph.cpp

```diff
@@ -344,31 +344,47 @@ GaussianFactorGraph::shared_ptr NonlinearFactorGraph::linearize(const Values& li
 }
 
 /* ************************************************************************* */
-static Scatter scatterFromValues(const Values& values, boost::optional<Ordering&> ordering) {
+static Scatter scatterFromValues(const Values& values) {
   gttic(scatterFromValues);
 
   Scatter scatter;
   scatter.reserve(values.size());
 
-  if (!ordering) {
-    // use "natural" ordering with keys taken from the initial values
-    for (const auto& key_value : values) {
-      scatter.add(key_value.key, key_value.value.dim());
-    }
-  } else {
-    // copy ordering into keys and lookup dimension in values, is O(n*log n)
-    for (Key key : *ordering) {
-      const Value& value = values.at(key);
-      scatter.add(key, value.dim());
-    }
+  // use "natural" ordering with keys taken from the initial values
+  for (const auto& key_value : values) {
+    scatter.add(key_value.key, key_value.value.dim());
+  }
+
+  return scatter;
+}
+
+/* ************************************************************************* */
+static Scatter scatterFromValues(const Values& values, const Ordering& ordering) {
+  gttic(scatterFromValues);
+
+  Scatter scatter;
+  scatter.reserve(values.size());
+
+  // copy ordering into keys and lookup dimension in values, is O(n*log n)
+  for (Key key : ordering) {
+    const Value& value = values.at(key);
+    scatter.add(key, value.dim());
   }
 
   return scatter;
 }
 
 /* ************************************************************************* */
 HessianFactor::shared_ptr NonlinearFactorGraph::linearizeToHessianFactor(
-    const Values& values, boost::optional<Ordering&> ordering, const Dampen& dampen) const {
+    const Values& values, const Dampen& dampen) const {
```

Comment:
potentially breaks code,
~~`linearizeToHessianFactor(values, boost::none, dampen)`~~ changes to
`linearizeToHessianFactor(values, dampen)`